### PR TITLE
[Issue 1056] Adds step by step release instructions

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary
Fixes https://github.com/HHS/simpler-grants-gov/issues/1056

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -88,7 +88,19 @@ This project takes a very collaborative and [agile](https://agilemanifesto.org/)
 
 Releases follow the [CalVer](https://calver.org/) versioning using a `YYYY.MM.DD-N` format. Start with `-1` for a release, incrementing if there is another release on the same day.
 
-Releases should be [created in Github](https://github.com/HHS/simpler-grants-gov/releases) and include a log of changes.
+Releases are [created in Github](https://github.com/HHS/simpler-grants-gov/releases) and with a log of changes.
+
+Step by step instructions for creating a release:
+
+- In your terminal, `git switch main` and `git pull` to sync your local code with the latest main commit
```
Comment (SammySteiner) [inline DEVELOPMENT.md:95]: I've still been using `checkout` but recently learned about `switch`! I appreciate the inclusion here

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -88,7 +88,19 @@ This project takes a very collaborative and [agile](https://agilemanifesto.org/)
...
+- In your terminal, `git tag YYYY.MM.DD-N` (FILLED IN!) and `git push --tags` to tag and push your CalVer tag
```
Comment (SammySteiner) [inline DEVELOPMENT.md:96]: Let's describe here how `N` is determined

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -88,7 +88,19 @@ This project takes a very collaborative and [agile](https://agilemanifesto.org/)
...
+- On Github.com, open the Releases page: https://github.com/HHS/simpler-grants-gov/releases
+- Click "Draft a new release"
+- Choose the tag you just pushed as the current tag, and the last CalVer release as the previous tag
+- Click "Generate release notes" to auto-generate release notes
+- Click "Publish" to finalize this step
+- On Github.com, open the Actions page: https://github.com/HHS/simpler-grants-gov/actions
+- You should see Actions that correspond to your release. Follow their status to ensure that they succeed. Follow-up may be required if the Github Actions fail.
```
Comment (SammySteiner) [inline DEVELOPMENT.md:103]: Let's add a link here to the github actions page for deploying a release

**Review by SammySteiner — APPROVED** [review]
Added a few suggestions, but looks good!
Thanks for making our docs better! 🎉

File: DEVELOPMENT.md (hunk 3 above)
Comment (chouinar) [inline DEVELOPMENT.md:103]: Not a blocker for this ticket, but a question - Do releases start deployments for the front-end, and API at the same time? That feels potentially risky.

**Review by chouinar — APPROVED** [review]
LGTM (+1 to each of Sammy's notes) with a related question inline

File: DEVELOPMENT.md (hunk 3 above)
Comment (coilysiren) [reply DEVELOPMENT.md:103]: @chouinar from looking at [a recent deploy](https://github.com/HHS/simpler-grants-gov/actions/runs/7758241676) plus the two config files [1](https://github.com/HHS/simpler-grants-gov/blob/main/.github/workflows/cd-api.yml), [2](https://github.com/HHS/simpler-grants-gov/blob/main/.github/workflows/cd-frontend.yml), it definitely looks like they deploy at exactly the same time.

It makes sense that we would want the backend API to always deploy first, right?

Comment (coilysiren) [reply DEVELOPMENT.md:103]: @SammySteiner I'm not 100% up to speed on how to Github Actions work yet, so I'm going to link the general Github actions page!

File: DEVELOPMENT.md (hunk 2 above)
Comment (coilysiren) [reply DEVELOPMENT.md:96]: @SammySteiner I moved the existing description and expanded it:

> Start with `N=1` for a release, incrementing if there is another release on the same day. So the first release on `2000.10.10` would be `2000.10.10-1`, ths second release on that day would be `2000.10.10-2`, etc.

File: DEVELOPMENT.md (hunk 3 above)
Comment (coilysiren) [reply DEVELOPMENT.md:104]: 🤦🏼‍♀️ I already did that. Please hold, I will figure out the specific Github Actions page for deploying a release 👍🏼

**Comment by coilysiren** [comment]
Gonna use these today! 🎉

**Pull request merged**